### PR TITLE
Add unit to RuntimeMetric

### LINKED ISSUE #17633: Add unit to RuntimeMetric
RuntimeMetric is used to track the per query execution stats. It was added in this issue: https://github.com/prestodb/presto/issues/16654

Currently it can only be used to track integer metrics, such as latency, CPU time, number of bytes, count, etc. There is currently no clean way to distinguish which type it is. It is better to add a enum unit to it so that we can display the metrics in different formats in the coordinator UI.

https://github.com/facebookincubator/velox/pull/1391 added the support in Velox. This issue tracks the work in the presto Java stack.

### PR DESCRIPTION
**Pull request by zacw7**
Resolves #17633 

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    // Time between task creation and start.
+    public static final RuntimeMetricKey TASK_QUEUED_TIME_NANOS = new RuntimeMetricKey("taskQueuedTimeNanos");
+    // Total operation time of a task on a worker. TASK_ELAPSED_TIME_NANOS - TASK_SCHEDULED_TIME_NANOS is the time when the task is doing nothing, e.g. it might be waiting for splits/inputs.
+    public static final RuntimeMetricKey TASK_SCHEDULED_TIME_NANOS = new RuntimeMetricKey("taskScheduledTimeNanos");
+    // Blocked time for the operators due to waiting for inputs.
+    public static final RuntimeMetricKey TASK_BLOCKED_TIME_NANOS = new RuntimeMetricKey("taskBlockedTimeNanos");
+    // Time taken for a read call to storage
+    public static final RuntimeMetricKey STORAGE_READ_TIME_NANOS = new RuntimeMetricKey("storageReadTimeNanos");
+    // Size of the data retrieved by read call to storage
+    public static final RuntimeMetricKey STORAGE_READ_DATA_BYTES = new RuntimeMetricKey("storageReadDataBytes");
```
Comment (NikhilCollooru) [inline presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java:60]: Why are we creating a RuntimeMetricKey with NONE as RunTimeUnit ? We know its BYTES right ?

File: presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        this(name, NONE);
+    }
+
+    @JsonCreator
+    @ThriftConstructor
+    public RuntimeMetricKey(@JsonProperty("name") String name, @JsonProperty("unit") RuntimeUnit unit)
+    {
+        this.name = name;
+        this.unit = unit;
```
Comment (NikhilCollooru) [inline presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java:75]: add `requireNonNull`

File: presto-main/src/main/java/com/facebook/presto/operator/Driver.java [hunk 3]
```diff
@@ -266,7 +266,7 @@ private void processNewSources()
                         .getFragmentResultCacheManager()
                         .get(fragmentResultCacheContext.get().get().getHashedCanonicalPlanFragment(), split);
                 sourceOperator.getOperatorContext().getRuntimeStats().addMetricValue(
-                        pages.isPresent() ? RuntimeMetricName.FRAGMENT_RESULT_CACHE_HIT : RuntimeMetricName.FRAGMENT_RESULT_CACHE_MISS, 1);
+                        pages.isPresent() ? RuntimeMetricKey.FRAGMENT_RESULT_CACHE_HIT : RuntimeMetricKey.FRAGMENT_RESULT_CACHE_MISS, 1);
```
Comment (NikhilCollooru) [inline presto-main/src/main/java/com/facebook/presto/operator/Driver.java:269]: lets static import and directly use ?

File: presto-common/src/main/java/com/facebook/presto/common/RuntimeUnit.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+@ThriftEnum
+public enum RuntimeUnit
+{
+    NONE(0), NANO(1), BYTE(2);
+
+    private final int value;
+
+    RuntimeUnit(int value)
+    {
+        this.value = value;
```
Comment (NikhilCollooru) [inline presto-common/src/main/java/com/facebook/presto/common/RuntimeUnit.java:28]: `requireNonNull`

File: presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java (hunk 1 above)
Comment (zacw7) [reply presto-common/src/main/java/com/facebook/presto/common/RuntimeMetricKey.java:60]: Another PR for setting the metrics is planned.

File: presto-common/src/main/java/com/facebook/presto/common/RuntimeUnit.java (hunk 4 above)
Comment (zacw7) [reply presto-common/src/main/java/com/facebook/presto/common/RuntimeUnit.java:28]: This parameter is `int` and cannot be `null` right?

Comment (NikhilCollooru) [reply presto-common/src/main/java/com/facebook/presto/common/RuntimeUnit.java:28]: missed that .. you are right !

**Review by highker — APPROVED** [review]

**Comment by aweisberg** [comment]
I think this may have broken verifier because because `RuntimeUnit` is not JSON serializable?
https://www.internalfb.com/intern/presto/verifier/result/?event_uuid=c7e7fb42-c685-4ae8-b74f-be40e4d0965d

**Comment by NikhilCollooru** [comment]
> I think this may have broken verifier because because `RuntimeUnit` is not JSON serializable? https://www.internalfb.com/intern/presto/verifier/result/?event_uuid=c7e7fb42-c685-4ae8-b74f-be40e4d0965d
> 
> Can we revert this until it is fixed to unblock releasing verifier fixes? We have release blocking fixes we are trying to get out.

yeah lets revet this and merge after we fix & test it.

**Comment by zacw7** [comment]
#17733 to revert this commit

**Comment by zacw7** [comment]
> I think this may have broken verifier because because `RuntimeUnit` is not JSON serializable? https://www.internalfb.com/intern/presto/verifier/result/?event_uuid=c7e7fb42-c685-4ae8-b74f-be40e4d0965d
> 
> Can we revert this until it is fixed to unblock releasing verifier fixes? We have release blocking fixes we are trying to get out.

I think the root cause is the old verifier cannot recognize the new added field. As long as there is inconsistency it is always gonna break. How do we handle this case?

**Comment by amitkdutta** [comment]
> > I think this may have broken verifier because because `RuntimeUnit` is not JSON serializable? https://www.internalfb.com/intern/presto/verifier/result/?event_uuid=c7e7fb42-c685-4ae8-b74f-be40e4d0965d
> > Can we revert this until it is fixed to unblock releasing verifier fixes? We have release blocking fixes we are trying to get out.
> 
> I think the root cause is the old verifier cannot recognize the new added field. As long as there is inconsistency it is always gonna break. How do we handle this case?

Breaking change is fine. As long a we can serialize/deserialize in json, it's good.

**Pull request merged**